pylint/checkers/utils.py

```diff
@@ -852,3 +852,20 @@ def is_registered_in_singledispatch_function(node):
             return decorated_with(func_def, singledispatch_qnames)
 
     return False
+
+
+def infer_sequence_with_starred_expressions(seq_node):
```

Comment:
@PCManticore only if we can figure out reasonable API. 

I did not had a good idea on how to do it "magically" in astroid (and I don't want to expand 'elts' implicitly, since effectively inferable and uninferable starred expressions are handled inconsistently), so creating helper function in pylint effectively "postponed" this difficult decision further into future.